src/tests/roc_library/test_sender_receiver.cpp

```diff
@@ -11,534 +11,282 @@
-            source_queue_.write(pp);
-        } else {
-            pp->udp()->dst_addr = receiver_repair_endp_;
-            repair_queue_.write(pp);
-        }
+    init_config(Flags);
 
-        for (;;) {
-            const size_t block_pos = pos_ % (n_source_packets_ + n_repair_packets_);
-
-            if (block_pos < n_source_packets_) {
-                if (!send_packet_(source_queue_, block_pos == 1)) {
-                    return;
-                }
-            } else {
-                if (!send_packet_(repair_queue_, false)) {
-                    return;
-                }
-            }
-        }
-    }
+    test::Context recv_context, send_context;
 
-    bool send_packet_(packet::IReader& reader, bool drop) {
-        packet::PacketPtr pp = reader.read();
-        if (!pp) {
-            return false;
-        }
-        pos_++;
-        if (!drop) {
-            writer_->write(pp);
-        }
-        return true;
-    }
+    test::Receiver receiver(recv_context, receiver_conf, sample_step, test::FrameSamples,
+                            Flags);
 
-    netio::UdpSenderConfig send_config_;
+    test::Sender sender(send_context, sender_conf, receiver.source_endpoint(),
+                        receiver.repair_endpoint(), sample_step, test::FrameSamples,
+                        Flags);
 
-    roc_endpoint* input_source_endp_;
-    roc_endpoint* input_repair_endp_;
+    sender.start();
+    receiver.run();
+    sender.stop();
+    sender.join();
+}
 
-    netio::UdpReceiverConfig recv_source_config_;
-    netio::UdpReceiverConfig recv_repair_config_;
+TEST(sender_receiver, multiple_senders_one_receiver) {
+    enum { Flags = 0 };
 
-    address::SocketAddr receiver_source_endp_;
-    address::SocketAddr receiver_repair_endp_;
+    init_config(Flags);
 
-    packet::Queue source_queue_;
-    packet::Queue repair_queue_;
+    test::Context context;
 
-    packet::IWriter* writer_;
+    test::Receiver receiver(context, receiver_conf, sample_step, test::FrameSamples,
+                            Flags);
 
-    netio::EventLoop event_loop_;
+    test::Sender sender_1(context, sender_conf, receiver.source_endpoint(),
+                          receiver.repair_endpoint(), sample_step, test::FrameSamples,
+                          Flags);
 
-    const size_t n_source_packets_;
-    const size_t n_repair_packets_;
+    sender_1.start();
+    receiver.run();
+    sender_1.stop();
+    sender_1.join();
 
-    size_t pos_;
-};
+    test::Sender sender_2(context, sender_conf, receiver.source_endpoint(),
+                          receiver.repair_endpoint(), sample_step, test::FrameSamples,
+                          Flags);
 
-} // namespace
+    sender_2.start();
+    receiver.run();
+    sender_2.stop();
+    sender_2.join();
+}
 
-TEST_GROUP(sender_receiver) {
-    roc_sender_config sender_conf;
-    roc_receiver_config receiver_conf;
+TEST(sender_receiver, multiple_receivers_and_senders) {
```

Comment:
The same concern with this test. We need a parallel version. Not sure if we need a sequential version of this test, given that we will already have a sequential version of the previous test. Up to you.